cassandra-executor/src/main/java/com/mesosphere/dcos/cassandra/executor/backup/AzureStorageDriver.java

```diff
@@ -0,0 +1,303 @@
+      IOUtils.closeQuietly(pageBlobOutputStream);
+    }
+  }
+
+  @Override
+  public void download(RestoreContext ctx) throws IOException {
+
+    final String accountName = ctx.getAcccountId();
+    final String accountKey = ctx.getSecretKey();
+    final String localLocation = ctx.getLocalLocation();
+    final String backupName = ctx.getName();
+    final String nodeId = ctx.getNodeId();
+
+    final String containerName = StringUtils.lowerCase(getContainerName(ctx.getExternalLocation()));
+    // https://<account_name>.blob.core.windows.net/<container_name>
+    final CloudBlobContainer container = getCloudBlobContainer(accountName, accountKey, containerName);
+
+    if (container == null) {
+      logger.error("Error uploading snapshots.  Unable to connect to {}, for container {}.",
+        ctx.getExternalLocation(), containerName, localLocation);
+      return;
+    }
+    String keyPrefix = String.format("%s/%s", backupName, nodeId);
+
+    final Map<String, Long> snapshotFileKeys = getSnapshotFileKeys(container, keyPrefix);
+    logger.info("Snapshot files for this node: {}", snapshotFileKeys);
+
+    for (String fileKey : snapshotFileKeys.keySet()) {
+      downloadFile(localLocation, container, fileKey, snapshotFileKeys.get(fileKey));
+    }
+  }
+
+  private void downloadFile(String localLocation, CloudBlobContainer container, String fileKey, long originalSize) {
+
+    logger.info("Downloading |  Local location {} | fileKey: {} | Size: {}", localLocation, fileKey, originalSize);
+
+    final String fileLocation = localLocation + File.separator + fileKey;
+    File file = new File(fileLocation);
+    // Only create parent directory once, if it doesn't exist.
+    if (!createParentDir(file)) {
+      logger.error("Unable to create parent directories!");
+      return;
+    }
+
+    InputStream inputStream = null;
+    SnappyInputStream compress = null;
+
+    try (
+      FileOutputStream fileOutputStream = new FileOutputStream(file, true);
+      BufferedOutputStream bos = new BufferedOutputStream(fileOutputStream)) {
+
+      final CloudPageBlob pageBlobReference = container.getPageBlobReference(fileKey);
+      inputStream = new PageBlobInputStream(pageBlobReference);
+      compress = new SnappyInputStream(inputStream);
+
+      IOUtils.copy(compress, bos, DEFAULT_PART_SIZE_DOWNLOAD);
+
+    } catch (Exception e) {
+      logger.error("Unable to write file: {}", fileKey, e);
+    } finally {
+      IOUtils.closeQuietly(compress);
+      IOUtils.closeQuietly(inputStream);
+    }
+  }
+
+  private String getContainerName(String externalLocation) {
+    return externalLocation.substring("azure://".length()).replace("/", "");
+  }
+
+  private CloudBlobContainer getCloudBlobContainer(String accountName, String accountKey, String containerName) {
+    CloudBlobContainer container = null;
+
+    if (StringUtils.isNotBlank(containerName)) {
+      final String storageConnectionString = "DefaultEndpointsProtocol=http"
```

Comment:
Can this ever be `https` ? If yes, is there any reason to not make it https by default ?